conf/modules.config

```diff
@@ -1,153 +1,70 @@
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_FAIDX {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_SORT {
-        ext.prefix = { "${meta.id}.sorted" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_INDEX {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_FLAGSTAT {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: SAMTOOLS_IDXSTATS {
-        ext.prefix = { "${meta.id}" }
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: FASTQC {
-        ext.args = '--quiet'
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: CAT_FASTQ {
-        publishDir = [
-            enabled: false
-        ]
-    }
-    withName: FASTP {
-        publishDir = [
-            enabled: false
+            path: "${params.outdir}",
+            mode: params.publish_dir_mode
         ]
     }
-    withName: MULTIQC {
-        ext.args   = { params.multiqc_title ? "--title \"$params.multiqc_title\"" : '' }
+
+    withName: 'GRZQC:PREPARE_REFERENCES:(SAMTOOLS_BGZIP|BWAMEM2_INDEX)' {
         publishDir = [
-            path: { "${params.outdir}/multiqc" },
+            path: { "${params.reference_path}/${meta.id}" },
             mode: params.publish_dir_mode,
+            enabled: "${params.reference_path != null}",
             saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
     }
+
     withName: PICARD_ADDORREPLACEREADGROUPS {
-        scratch = true
-        ext.args =  { [
-            "--VALIDATION_STRINGENCY LENIENT",
-            "--RGLB lib",
+        ext.prefix = { "${meta.id}.rrg" }
+        ext.args = { [
+            "--RGLB library",
             "--RGPL platform",
             "--RGPU barcode",
             "--RGSM ${meta.id}",
-            "--TMP_DIR ."
-            ].join(' ').trim() }
-        publishDir = [
-            enabled: false
-        ]
+            "--TMP_DIR ptmp/"
+        ].join(' ').trim() }
     }
+
     withName: PICARD_MARKDUPLICATES {
-        scratch = true
         ext.prefix = { "${meta.id}.md" }
-        ext.args   = { [
+        ext.args = { [
             "--CREATE_INDEX",
-            "--TMP_DIR ."
-            ].join(' ').trim() }
-        publishDir = [
-            enabled: false
-        ]
+            "--TMP_DIR ptmp/"
+        ].join(' ').trim() }
     }
+
     withName: MOSDEPTH {
-        ext.prefix = { "${meta.id}" }
         publishDir = [
             path:"${params.outdir}/mosdepth/",
             pattern: "*{.bed.gz,.summary.txt}",
             mode: params.publish_dir_mode
         ]
     }
-    withName: CONVERT_BED_CHROM {
```

Comment:
why this is deleted :D we are not using the module anymore? nextflow automatically creates output dirs if publishDir is not disabled.